frappe/model/workflow.py

```diff
@@ -98,6 +98,15 @@ def is_transition_condition_satisfied(transition, doc) -> bool:
 		return frappe.safe_eval(transition.condition, get_workflow_safe_globals(), dict(doc=doc.as_dict()))
 
 
+def evaluate_workflow_value(value, doc):
+	if not value:
+		return None
+	try:
+		return frappe.safe_eval(value, get_workflow_safe_globals(), dict(doc=doc.as_dict()))
+	except Exception:
```

Comment:
This is to handle non-expressions I guess?

Seems a bit weird to handle this implicitly, an invalid expression would become a value too. 

I prefer having a checkbox or some way for the user to decide whether they want a value or an expression here.